app/models/service_template_transformation_plan/validate_config_info.rb

```diff
@@ -63,6 +63,7 @@ def validate_config_info(options)
           raise _("Invalid VM found #{vm_obj.name}") if vm_obj.invalid?
 
           vm_options = {}
+          vm_options[:warm_migration_compatible] = vm_obj.snapshots.blank?
```

Comment:
This assumes that having snapshots is the only thing that would prevent warm migration which might not always be the case.  How about a supports feature mixin on the (I assume only VMware VM) model?  Then this becomes `vm_object.supports_warm_migration?`